src/services/decoratelayerloading.js

```diff
@@ -0,0 +1,79 @@
+goog.provide('ngeo.DecorateLayerLoading');
+
+goog.require('goog.asserts');
+goog.require('ngeo');
+
+
+/**
+ * Provides a function that adds a 'loading 'property (using
+ * `Object.defineProperty`) to the layer.
```

Comment:
Do you maybe want to mention that the service is only supposed to be used for layer with tile and image sources?